src/elysium_connection.erl

```diff
@@ -170,20 +171,53 @@ with_connection(Config, Mod, Fun, Args, Consistency)
+                false -> with_connection(Config, Mod, Fun, Args, Consistency, Before_Checkout_Connection, Cmd_Details);
+                true  ->
+                    After_Checkout_Connection = erlang:now(),
+                    Checkout_Connection_Duration =timer:now_diff(After_Checkout_Connection, Before_Checkout_Connection),
+                    elysium_prometheus:report_metrics(cassandra_get_connection_duration_microseconds, Cmd_Details, Checkout_Connection_Duration),
+                    Reply_Timeout = elysium_config:request_reply_timeout(Config),
+                    Query_Request = {mod_fun, Config, Mod, Fun, Args, Consistency},
+                    Before_Exec_Cmd = erlang:now(),
+                    Reply = elysium_buffering_strategy:handle_pending_request(Config, BS_Module, 0, Reply_Timeout, Node, Sid, Query_Request),
+                    After_Exec_Cmd = erlang:now(),
+                    Exec_Cmd_Duration = timer:now_diff(After_Exec_Cmd, Before_Exec_Cmd),
+                    elysium_prometheus:report_metrics(cassandra_exec_cmd_duration_microseconds, Cmd_Details, Exec_Cmd_Duration),
+                    handle_mod_fun_reply(Buffering_Strategy, Reply, Mod, Fun, Args)
             end
     end.
 
+with_connection(Config, Mod, Fun, Args, Consistency, Before_Checkout_Connection, Cmd_Details)
+    when is_atom(Mod), is_atom(Fun), is_list(Args) ->
+    true = erlang:function_exported(Mod, Fun, 3),
+    {Buffering_Strategy, BS_Module} = get_buffer_strategy_module(Config),
+    case elysium_buffering_strategy:checkout_connection(Config) of
+        none_available ->
+            buffer_mod_fun_call(Config, Mod, Fun, Args, Consistency, Buffering_Strategy);
+        {Node, Sid} when is_pid(Sid) ->
+            case is_process_alive(Sid) of
+                false -> with_connection(Config, Mod, Fun, Args, Consistency);
+                true  ->
+                    After_Checkout_Connection = erlang:now(),
+                    Checkout_Connection_Duration = timer:now_diff(After_Checkout_Connection, Before_Checkout_Connection),
+                    elysium_prometheus:report_metrics(cassandra_get_connection_duration_microseconds, Cmd_Details, Checkout_Connection_Duration),
+                    Reply_Timeout = elysium_config:request_reply_timeout(Config),
+                    Query_Request = {mod_fun, Config, Mod, Fun, Args, Consistency},
+                    Before_Exec_Cmd = erlang:now(),
+                    Reply = elysium_buffering_strategy:handle_pending_request(Config, BS_Module, 0, Reply_Timeout, Node, Sid, Query_Request),
+                    After_Exec_Cmd = erlang:now(),
+                    Exec_Cmd_Duration = timer:now_diff(After_Exec_Cmd, Before_Exec_Cmd),
+                    elysium_prometheus:report_metrics(cassandra_exec_cmd_duration_microseconds, Cmd_Details, Exec_Cmd_Duration),
+                    handle_mod_fun_reply(Buffering_Strategy, Reply, Mod, Fun, Args)
```

Comment:
maybe better to move those report_metrics functions after handle_mod_fun_reply and put the real reporing functions in a gen_server or seperate process? 
consider this function will be a really hotspot in every scenario,  small delay will introduce a big impact on the performance. 
So i will prefer to move all those parse query, report metrics functions into a seperate process or function.